integration/devmode/devmode_test.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright IBM Corp All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package devmode
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"syscall"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/hyperledger/fabric/integration/nwo"
+	"github.com/hyperledger/fabric/integration/nwo/commands"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+	"github.com/tedsuo/ifrit"
+	"github.com/tedsuo/ifrit/ginkgomon"
+)
+
+var _ = Describe("Devmode", func() {
+	var (
+		testDir                  string
+		client                   *docker.Client
+		network                  *nwo.Network
+		process                  ifrit.Process
+		chaincode                nwo.Chaincode
+		chaincodeRunner          *ginkgomon.Runner
+		chaincodeProcess         ifrit.Process
+		channelName, chaincodeID string
+	)
+
+	BeforeEach(func() {
+		var err error
+		channelName = "testchannel"
+		testDir, err = ioutil.TempDir("", "devmode")
+		Expect(err).NotTo(HaveOccurred())
+
+		client, err = docker.NewClientFromEnv()
+		Expect(err).NotTo(HaveOccurred())
+
+		chaincode = nwo.Chaincode{
+			Name:    "mycc",
+			Version: "0.0",
+			Path:    "github.com/hyperledger/fabric/integration/chaincode/simple/cmd",
+			Ctor:    `{"Args":["init","a","100","b","200"]}`,
+			Policy:  `AND ('Org1MSP.member','Org2MSP.member')`,
+		}
+
+		chaincodeID = fmt.Sprintf("%s:%s", chaincode.Name, chaincode.Version)
+		network = nwo.New(nwo.BasicSolo(), testDir, client, BasePort(), components)
+		network.TLSEnabled = false
+		// turn on dev mode for all peers
+		for _, p := range network.Peers {
+			p.DevMode = true
+		}
+
+		network.GenerateConfigTree()
+		network.Bootstrap()
+
+		networkRunner := network.NetworkGroupRunner()
+		process = ifrit.Invoke(networkRunner)
+		Eventually(process.Ready(), network.EventuallyTimeout).Should(BeClosed())
+	})
+
+	AfterEach(func() {
+		if process != nil {
+			process.Signal(syscall.SIGTERM)
+			Eventually(process.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+
+		if chaincodeProcess != nil {
+			chaincodeProcess.Signal(syscall.SIGTERM)
+		}
+
+		if network != nil {
+			network.Cleanup()
+		}
+
+		os.RemoveAll(testDir)
+	})
+
+	It("executes chaincode in dev mode", func() {
+		org1peer0 := network.Peer("Org1", "peer0")
+		orderer := network.Orderer("orderer")
+
+		By("setting up the channel")
+		network.CreateAndJoinChannel(orderer, channelName)
+
+		By("building the chaincode")
+		cmd, err := gexec.Build(chaincode.Path)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("running the chaincode")
+		peerChaincodeAddress := network.PeerAddress(org1peer0, nwo.ChaincodePort)
+		chaincodeRunner = network.ChaincodeRunner(cmd, buildPeerEnv(peerChaincodeAddress, chaincodeID)...)
+		chaincodeProcess = ifrit.Invoke(chaincodeRunner)
+		Eventually(chaincodeProcess.Ready(), network.EventuallyTimeout).Should(BeClosed())
```

Comment:
I'm confused. We're running a multi-peer network, starting an instance of the chaincode, and pointing it to a peer. Later, we install and instantiate the chaincode across two peers.

What asserts that our "dev" chaincode is being used?
How is the "dev" chaincode possibly getting used by the second instance of the peer given we've started it with a reference to the first?
How common is it for "dev" mode to require multiple endorsements?

Reply:
🧹 
- Reduced the scope to a single peer and modified the endorsement policy for the chaincode
- Asserting the "dev" chaincode is being used by confirming no docker containers exist during the first query